app/bulk_resource/views.py

```diff
@@ -22,33 +32,115 @@ def index():
             csv_container = CsvContainer(
                 date_uploaded=datetime.now(),
                 file_name=csv_data.filename,
+                user=current_user
             )
-            # TODO: Associate CSV file with current_user.
-            # csv_container.user = current_user
+            csv_reader = csv.reader(csv_file)
+
+            # The first row of the CSV file contains the names of the columns.
+            header_row = csv_reader.next()
+            csv_header_row = CsvHeaderRow(
+                csv_header_row_container=csv_container
+            )
+            for column_name in header_row:
+                csv_cell = CsvHeaderCell(data=column_name,
+                                         csv_header_row=csv_header_row)
+                db.session.add(csv_cell)
+            db.session.add(csv_header_row)
 
             # Iterate through the CSV file row-by-row and then cell-by-cell.
             # Each cell contains one comma-separated string in a row of a CSV
             # file.
-            csv_reader = csv.reader(csv_file)
             for row in csv_reader:
-                csv_row = CsvRow(csv_container=csv_container)
+                csv_body_row = CsvBodyRow(csv_container=csv_container)
                 for cell_data in row:
-                    csv_cell = CsvCell(data=cell_data, csv_row=csv_row)
+                    csv_cell = CsvBodyCell(data=cell_data,
+                                           csv_body_row=csv_body_row)
                     db.session.add(csv_cell)
-                db.session.add(csv_row)
+                db.session.add(csv_body_row)
             db.session.add(csv_container)
             db.session.commit()
-
             # TODO: Error catching if CSV is malformed.
-
-        return redirect(url_for('bulk_resource.review'))
+            # TODO: Check that CSV file has "Name", "Address" headings
+        return redirect(url_for('bulk_resource.review1'))
     return render_template('bulk_resource/upload.html', form=form)
 
 
-@bulk_resource.route('/review')
+@bulk_resource.route('/review1', methods=['GET', 'POST'])
 @login_required
-def review():
-    # TODO: Retrieve CSV associated with current_user.
-    csv_container = CsvContainer.query.first()
-    return render_template('bulk_resource/review.html',
-                           csv_container=csv_container)
+def review1():
```

Comment:
Could you make more descriptive URLs and function names? How about something like this:
/review-descriptor-types, review_descriptor_types
/review-options, review_options
?